packages/grafana-ui/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FunctionComponent } from 'react';
+import { Range, createSliderWithTooltip } from 'rc-slider';
+import { css } from 'emotion';
+import { stylesFactory } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
+import { useTheme } from '../../themes/ThemeContext';
+
+export enum SliderOrientation {
```

Comment:
I've created a very similar type yesterday (https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Layout/Layout.tsx#L6) but I'm not happy with it. Can you please unify these two types to a single one and place in grafana-ui/types?

Also, I have second thoughts about using `enum`. I know it's tempting, but it has some consequences later on when consuming the API. I.e. you cannot use string literals when applying the component:

```tsx
<Slider orientation="horizontal" ... />
```

will not work (technically it will, but TS will complain), and in order to make it work you will need to import the `SliderOrientation` enum from grafana-ui too:

```tsx
import { Slider, SliderOrientation } from '@grafana/ui';

<Slider orientation={SliderOrientation.horizontal} />
```

Instead just define type:  
```ts
type Orientation = 'horizontal' | 'vertical';
```

---
I'd rather avoid the need to import enums from @grafana/ui and I'd like above(type instead of enum) to become a pattern we use. Some thougts on this @torkelo ?

Reply:
Yea, enums can be good but for props like this I find a string type like:
type Orientation = 'horizontal' | 'vertical';

better as you can use the component without importing more types.